module/Application/view/error/404.phtml

```diff
@@ -1,107 +1,103 @@
-<h1><?php echo $this->translate('A 404 error occurred') ?></h1>
-<h2><?php echo $this->message ?></h2>
-
-<?php if (isset($this->reason) && $this->reason): ?>
+<h1>A 404 error occurred</h1>
+<h2><?= $this->message ?></h2>
 
 <?php
-$reasonMessage= '';
-switch ($this->reason) {
-    case 'error-controller-cannot-dispatch':
-        $reasonMessage = $this->translate('The requested controller was unable to dispatch the request.');
-        break;
-    case 'error-controller-not-found':
-        $reasonMessage = $this->translate('The requested controller could not be mapped to an existing controller class.');
-        break;
-    case 'error-controller-invalid':
-        $reasonMessage = $this->translate('The requested controller was not dispatchable.');
-        break;
-    case 'error-router-no-match':
-        $reasonMessage = $this->translate('The requested URL could not be matched by routing.');
-        break;
-    default:
-        $reasonMessage = $this->translate('We cannot determine at this time why a 404 was generated.');
-        break;
-}
+if (! empty($this->reason)) :
+
+    $reasonMessage= '';
+    switch ($this->reason) {
+        case 'error-controller-cannot-dispatch':
```

Comment:
Maybe we can use here constants?
I think once case is missing here.
Please have a look on my changes to zend-skeleton-application:
https://github.com/zendframework/ZendSkeletonApplication/pull/344/files#diff-162067fcf8cbfae3a0fc7ea971e15954R6